### ISSUE TITLE
Low BLEU scores with the almost baseline model NMT

### ISSUE THREAD

**Issue by LeenaShekhar** [OpenNMT/OpenNMT-py#553]
I trained a NMT model for De-En with default parameters except "global_attention" which I set to "mlp" during training. I also did not use "replace_unk" during decoding. 
Below are the commands that I used:

1. **Preprocessing**
<code>
python preprocess.py -train_src data/de-train-clean.200K.txt -train_tgt data/en-train-clean.200K.txt -valid_src data/de-val-clean.10K.txt -valid_tgt data/en-val-clean.10K.txt -save_data data/de-en-demo
</code>

2. **Training** (200k sentences De-En)
<code>
python train.py -data data/de-en-demo -save_model model-2-2/de-en-sgd-2-model -global_attention mlp --epochs 20
</code>

3. **Decoding**
<code>
python translate.py -model model-2-2/de-en-sgd-2-model_acc_45.85_ppl_29.23_e14.pt -src data/de-val-clean.10K.txt -tgt data/en-val-clean.10K.txt -output de-en-val-pred-10K.txt 
</code>

4. **Evaluation**
<code>
perl tools/multi-bleu.perl data/en-val-clean.10K.txt < de-en-val-10K-pred.txt
</code>

On the validation set I got the following scores:
BLEU = 14.15, 48.5/20.3/10.0/5.1 (BP=0.944, ratio=0.946, hyp_len=124751, ref_len=131939)

On the Newstest dataset I got the following scores (without -replace_unk):
BLEU = 8.11, 35.3/13.2/5.5/2.3 (BP=0.925, ratio=0.927, hyp_len=52009, ref_len=56088)

On the Newstest dataset I got the following scores (with -replace_unk):
BLEU = 9.04, 37.1/14.4/6.2/2.8 (BP=0.925, ratio=0.927, hyp_len=52009, ref_len=56088)

My scores seem to be very low compared to the original reported scores. I don't think just changing attention type should have such a drastic effect. 

Peplexity does not go down to around 20 as reported in the original 200k model in the README. Am I missing something? 

My training seems something like this with each epoch:
![image](https://user-images.githubusercontent.com/12227436/35827285-64d3073e-0a89-11e8-9ef7-5de364e4b2c7.png)

**Comment by srush** [comment]
Thank you for this report. We are looking into it immediately. Trying to reproduce on my end. 

If you could send your full log that would be great.

**Comment by srush** [comment]
Oh I apologize, the documentation is misleading. 

For En->De we get around 20 ppl. For De ->En, 29 ppl for that data is similar to our results (and also lua OpenNMT ).

This dataset is not meant to be representative. It is simply a test we use. It is much too small for use as a benchmark. 

If you want a small data set, I would recommend using IWSLT.

**Comment by LeenaShekhar** [comment]
Just to be clear, on the 200k dataset even your score was around 29? If that is the case then it makes sense. I will train it with IWSLT. Thank you.

**Comment by srush** [comment]
Yes, here's an earlier log for that data:
https://github.com/OpenNMT/IntegrationTesting/blob/master/logs/call_log_2016-12-03-16-17.txt

```
Epoch 8 ; Batch 3100/3149 ; LR 1.0000 ; Throughput 3170/1123/2047 total/src/targ tokens/sec ; PPL 18.36 ; Free mem 2062942208	
Validation PPL: 29.53171699102	
Saving checkpoint to demo-model_epoch8_29.53.t7..
```

I will update the documentation

**Comment by LeenaShekhar** [comment]
Thank you very much. Sorry about not checking that directory.

**Comment by srush** [comment]
Let's leave it open for now. I want to try to compare MLP and dot product attention to confirm.

**Comment by LeenaShekhar** [comment]
Sure. Once I am through training I will update as well in case this still remains open.

**Comment by srush** [comment]
After 9 Epochs on your data with default params, I am here:

```
Train perplexity: 15.7421
Train accuracy: 51.1333
Loading valid dataset from /n/home00/srush/projects/IntegrationTesting/data/demo2.valid.1.pt, number of examples: 5718
Validation perplexity: 28.1238
Validation accuracy: 46.564
Decaying learning rate to 0.25
```

I'll try with your attention (which should be better).

**Comment by LeenaShekhar** [comment]
That is weird. Mine is way below for accuracy and high on perplexity.
